File: mobile_app/lib/core/models/recipe.dart
class Recipe {
  Recipe({
    required this.id,
    required this.title,
    required this.category,
    required this.description,
    required this.ingredients,
    required this.instructions,
    required this.imageUrl,
    required this.averageRating,
  });

  final String id;
  final String title;
  final String category;
  final String description;
  final List<String> ingredients;
  final List<String> instructions;
  final String imageUrl;
  final double averageRating;

  factory Recipe.fromJson(Map<String, dynamic> json) {
    return Recipe(
      id: json['_id'] as String,
      title: json['title'] as String? ?? '',
      category: json['category'] as String? ?? '',
      description: json['description'] as String? ?? '',
      ingredients: (json['ingredients'] as List<dynamic>? ?? [])
          .map((item) => item.toString())
          .toList(),
      instructions: (json['instructions'] as List<dynamic>? ?? [])
          .map((item) => item.toString())
          .toList(),
      imageUrl: json['imageUrl'] as String? ?? '',
      averageRating: (json['averageRating'] as num? ?? 0).toDouble(),
    );
  }
}

File: backend/src/models/Recipe.js
const mongoose = require('mongoose')

const recipeSchema = new mongoose.Schema(
  {
    title: { type: String, required: true, trim: true },
    description: { type: String, default: '' },
    ingredients: [{ type: String, required: true }],
    instructions: [{ type: String, required: true }],
    category: { type: String, required: true, trim: true },
    imageUrl: { type: String, default: '' },
    prepTime: { type: Number, default: 0 },
    cookTime: { type: Number, default: 0 },
    servings: { type: Number, default: 1 },
    author: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    averageRating: { type: Number, default: 0 },
    ratingCount: { type: Number, default: 0 },
  },
  { timestamps: true },
)

module.exports = mongoose.model('Recipe', recipeSchema)
